cli/api/controller_client.go

```diff
@@ -0,0 +1,166 @@
+package api
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	api "github.com/amalgam8/amalgam8/cli/client"
+	"github.com/amalgam8/amalgam8/cli/common"
+	"github.com/amalgam8/amalgam8/cli/utils"
+	"github.com/urfave/cli"
+)
+
+type controller struct {
+	ctx    *cli.Context
+	client api.Client
+	debug  bool
+}
+
+// ControllerClient .
+type ControllerClient interface {
+	Routes() (*RouteList, error)
+	ServiceRoutes(service string) (*RuleList, error)
+	GetActions() (*ActionList, error)
+	Rules(query string) (*RuleList, error)
+	SetRules(payload io.Reader) (interface{}, error)
+	UpdateRules(payload io.Reader) (interface{}, error)
+	DeleteRules(query string) (interface{}, error)
+	NewQuery() url.Values
+}
+
+// NewControllerClient .
+func NewControllerClient(ctx *cli.Context) (ControllerClient, error) {
+	url, err := ValidateControllerURL(ctx)
+	if err != nil {
+		fmt.Fprintf(ctx.App.Writer, fmt.Sprintf("%s: %q\n\n", err.Error(), url))
+		return nil, err
+	}
+
+	// Read Token
+	token := ctx.GlobalString(common.ControllerToken.Flag())
+
+	// Check if a custom client has been set
+	var client *http.Client
+	if c, ok := ctx.App.Metadata["httpClient"].(*http.Client); ok {
+		client = c
+	}
+
+	return &controller{
+		debug:  ctx.GlobalBool(common.Debug.Flag()),
+		client: api.NewClient(url, token, client),
+	}, nil
+}
+
```

Comment:
No, it's not using the newly refactored controller client.

When I started working on the CLI, I wrote the controller and registry clients based on the idea that the new CLI was going to have its own repo, like the python version.

Now that I have updated the code to make the CLI part of the main A8 repo and added more commands, it makes more sense to reuse/update code from/to other packages so I will start looking at those changes.

Since we are releasing the CLI as beta version, I think is okay to merge the CLI as it is for now. We can remove the BETA tag once the CLI is more integrated with the existing code along and the corresponding doc updates and IT.